internal/handlers/common/compare.go

```diff
@@ -248,8 +249,23 @@ func compareOrdered[T constraints.Ordered](a, b T) compareResult {
 }
 
 // compareNumbers compares two numbers.
-//
-// TODO https://github.com/FerretDB/FerretDB/issues/371
 func compareNumbers(a float64, b int64) compareResult {
-	return compareOrdered(a, float64(b))
+	if math.IsNaN(a) {
```

Comment:
I replaced that new code with the old code (just `return compareOrdered(a, float64(b))`) and all tests still pass. We should write tests that fail with the old code to ensure that it is fixed

Reply:
I've pulled the changes from the master branch and all tests pass, do I need to add additional tests?

Reply:
> I've pulled the changes from the master branch and all tests pass, do I need to add additional tests?

Hope my comment would help you:

@DoodgeMatvey please see the issue https://github.com/FerretDB/FerretDB/issues/371 it describes the bug and it has also a playground link https://go.dev/play/p/tYJkQoHGr4I so it might be the test which is necessary to add, 

so on the main branch it fails and in this PR it is not and therefore it's obvious that the PR fixed bug.

Reply:
Thanks I got it. I added 1 additional case for `TestQueryComparisonEq`, `TestQueryComparisonGt`, `TestQueryComparisonLt`. I tested it and on the main branch it fails. but in this PR it is not